src/components/TokenAdministrative.js

```diff
@@ -98,34 +98,16 @@ class TokenAdministrative extends React.Component {
   /**
    * Update token state after didmount or props update
    */
-  updateWalletInfo = () => {
+  updateWalletInfo = async () => {
     if (!this.props.token && !this.props.token.uid) {
       return;
     }
 
-    const { uid } = this.props.token;
+    const mintUtxos = await this.props.wallet.getMintAuthority(this.props.token.uid, { many: true });
+    const meltUtxos = await this.props.wallet.getMeltAuthority(this.props.token.uid, { many: true });
 
-    let mintCount = 0;
-    let meltCount = 0;
-
-    const tokenBalance = this.props.tokensBalance[uid];
-
-    if (this.props.useWalletService) {
-      // Wallet Service
-      // TODO: This should come from the wallet service and display the count properly
-      const canMintUtxos = tokenBalance.mint;
-      const canMeltUtxos = tokenBalance.melt;
-
-      mintCount = canMintUtxos ? 1 : 0;
-      meltCount = canMeltUtxos ? 1 : 0;
-    } else {
-      // Old Facade
-      const mintUtxos = this.props.wallet.getMintAuthority(this.props.token.uid, { many: true });
-      const meltUtxos = this.props.wallet.getMeltAuthority(this.props.token.uid, { many: true });
-
-      mintCount = mintUtxos ? mintUtxos.length : 0;
-      meltCount = meltUtxos ? meltUtxos.length : 0;
-    }
+    const mintCount = mintUtxos ? mintUtxos.length : 0;
```

Comment:
Now that you changed the return to always return an array (empty when we were returning `null` before), then we can directly use here:

```
const mintCount = mintUtxos.length;
const meltCount = meltUtxos.length;
```

Reply:
Done in 3d1e2fa0cb04a84fc42213ed7c5e95e72f4969d6

Thanks